arelle/plugin/validate/ESEF_2022/__init__.py

```diff
@@ -0,0 +1,1225 @@
+                modelObject=contextsWithDisallowedOCEs, contextIds=", ".join(c.id for c in contextsWithDisallowedOCEs))
+        if contextsWithDisallowedOCEcontent:
+            modelXbrl.error("ESEF.2.1.3.scenarioContainsNonDimensionalContent",
+                _("xbrli:scenario in contexts MUST NOT contain any other content than defined in XBRL Dimensions specification: %(contextIds)s"),
+                modelObject=contextsWithDisallowedOCEcontent, contextIds=", ".join(c.id for c in contextsWithDisallowedOCEcontent))
+        if len(contextIdentifiers) > 1:
+            modelXbrl.error("ESEF.2.1.4.multipleIdentifiers",
+                _("All entity identifiers in contexts MUST have identical content: %(contextIds)s"),
+                modelObject=modelXbrl, contextIds=", ".join(i[1] for i in contextIdentifiers))
+        requiredScheme = val.authParam["identiferScheme"]
+        for (contextScheme, contextIdentifier), contextElts in contextIdentifiers.items():
+            if contextScheme != requiredScheme:
+                modelXbrl.warning("ESEF.2.1.1.nonLEIContextScheme" if requiredScheme == iso17442 else "UK.ESEF.2.1.1.contextScheme",
+                    _("The scheme attribute of the xbrli:identifier element should have \"%(requiredScheme)s\" as its content: %(contextScheme)s"),
+                    modelObject=contextElts, contextScheme=contextScheme, requiredScheme=requiredScheme)
+            elif contextScheme == iso17442:
+                leiValidity = LeiUtil.checkLei(contextIdentifier)
+                if leiValidity == LeiUtil.LEI_INVALID_LEXICAL:
+                    modelXbrl.error("ESEF.2.1.1.invalidIdentifierFormat",
+                        _("The LEI context identifier has an invalid format: %(identifier)s"),
+                        modelObject=contextElts, identifier=contextIdentifier)
+                elif leiValidity == LeiUtil.LEI_INVALID_CHECKSUM:
+                    modelXbrl.error("ESEF.2.1.1.invalidIdentifier",
+                        _("The LEI context identifier has checksum error: %(identifier)s"),
+                        modelObject=contextElts, identifier=contextIdentifier)
+        if contextsWithPeriodTime:
+            modelXbrl.error("ESEF.2.1.2.periodWithTimeContent",
+                _("The xbrli:startDate, xbrli:endDate and xbrli:instant elements MUST identify periods using whole days (i.e. specified without a time content): %(contextIds)s"),
+                modelObject=contextsWithPeriodTime, contextIds=", ".join(c.id for c in contextsWithPeriodTime))
+        if contextsWithPeriodTimeZone:
+            modelXbrl.error("ESEF.2.1.2.periodWithTimeZone",
+                _("The xbrli:startDate, xbrli:endDate and xbrli:instant elements MUST identify periods using whole days (i.e. specified without a time zone): %(contextIds)s"),
+                modelObject=contextsWithPeriodTimeZone, contextIds=", ".join(c.id for c in contextsWithPeriodTimeZone))
```

Comment:
Required for type checking/CI (this is in the other ESEF plugin):
```suggestion
                modelObject=contextsWithPeriodTimeZone, contextIds=", ".join(c.id for c in contextsWithPeriodTimeZone if c.id))
```